### ISSUE TITLE
unable to open install tool with incomplete localconfig.php

### ISSUE THREAD

**Issue by fritzmg** [contao/installation-bundle#16]
If your `/system/config/localconfig.php` is incomplete (for whatever reason) and you want to access the Install Tool (`/web/install.php`), the following error happens:

```
Fatal error: Uncaught exception 'Contao\CoreBundle\Exception\IncompleteInstallationException' with message 'The installation has not been completed. Open the Contao install tool to continue.'
```

The error message suggests to "Open the Contao install tool to continue.", however that's exactly what I tried to do ;)

The only way to get out of this is to completely delete the `localconfig.php`.

**Comment by fritzmg** [comment]
This is because [`InstallationKernel::boot`](https://github.com/contao/installation-bundle/blob/1.1.0-beta1/src/HttpKernel/InstallationKernel.php#L31) executes `InstallationKernel::bootRealSystem` once a `localconfig.php` is present and this in turn initializes the Contao framework:

``` php
$container->get('contao.framework')->initialize();
```

which in turn calls `ContaoFramework::validateInstallation` at some point, which throws this error.

**Comment by leofeyer** [comment]
What do you mean by "incomplete localconfig.php"?

**Comment by fritzmg** [comment]
> What do you mean by "incomplete localconfig.php"?

e.g. with either of these contents:

``` php
<?php
```

``` php
<?php

### INSTALL SCRIPT START ###
$GLOBALS['TL_CONFIG']['licenseAccepted'] = false;
### INSTALL SCRIPT STOP ###
```

``` php
<?php

### INSTALL SCRIPT START ###
$GLOBALS['TL_CONFIG']['licenseAccepted'] = false;
$GLOBALS['TL_CONFIG']['installPassword'] = '…';
### INSTALL SCRIPT STOP ###
```

``` php
<?php

### INSTALL SCRIPT START ###
$GLOBALS['TL_CONFIG']['installPassword'] = '…';
### INSTALL SCRIPT STOP ###
```

**Comment by leofeyer** [comment]
True, because that is actually how we check if the installation is complete:

https://github.com/contao/core-bundle/blob/master/src/Resources/contao/library/Contao/Config.php#L316

**Comment by fritzmg** [comment]
Yep I know. This is probably a benign issue, since there is not a valid reason to have a `localconfig.php` present with any of the content described above.

However, conceivably some user might think it's ok to delete the content of the `localconfig.php`, if he wants to reset his install tool password for instance, instead of just removing the respective entry - then it might be confusing to him what to do.

**Comment by leofeyer** [comment]
> some user might think it's ok to delete the content of the `localconfig.php`, if he wants to reset his install tool password for instance

They might think so, but they are still wrong. If you remove the entire local configuration, your installation is in fact no longer complete. Therefore it is correct to throw the exception in this case.

**Comment by fritzmg** [comment]
Yes, but the exception tells you to open the install tool, which is where the exception occurs.

**Comment by leofeyer** [comment]
Fixed in contao/core-bundle@6bb407c1db5ab6996fe39f1a286422a4a56dda8a.
